src/_pytest/fixtures.py

```diff
@@ -1245,21 +1217,57 @@ def __call__(self, function: FixtureFunction) -> FixtureFunction:
                 pytrace=False,
             )
 
-        # Type ignored because https://github.com/python/mypy/issues/2087.
-        function._pytestfixturefunction = self  # type: ignore[attr-defined]
-        return function
+        return fixture_definition
+
+
+# TODO: paramspec/return type annotation tracking and storing
+class FixtureFunctionDefinition:
+    def __init__(
+        self,
+        function: Callable[..., Any],
+        fixture_function_marker: FixtureFunctionMarker,
+        instance: type | None = None,
+    ):
+        self.name = fixture_function_marker.name or function.__name__
+        self.__name__ = self.name
```

Comment:
Can you add a comment on this line, explaining what it does? IIUC, it's to match the `__name__` attribute of function objects, but it's not obvious.

Reply:
Thanks, let me know if this is clear now.